### ISSUE TITLE
Consider consolidating Antigen/Fuzzlyn pipeline

### ISSUE THREAD

**Issue by kunalspathak** [dotnet/runtime#62107]
Today, there are 2 pipelines to conduct exploratory JIT testing: [Antigen](https://dev.azure.com/dnceng/public/_build?definitionId=1049&_a=summary) and [Fuzzlyn](https://dev.azure.com/dnceng/public/_build?definitionId=1054) . Most often during PR reviews, one of the two pipeline is triggered. Since both tools have an ability to catch unique issues, ideally both the pipelines should be triggered. In future, we even would like to trigger these pipelines on every PR. As such, we should consider unifying these two pipelines and just called them "exploratory" pipeline or some other better name. When someone triggers this pipeline, it will run both the tools Antigen/Fuzzlyn and report back the result.

Turns out this can be easily achieved because most of the pipeline code is shared between Fuzzlyn/Antigen pipelines. The only blocker currently is constructing the "Extensions" markdown page. It is created once the pipeline is finished. If we unify the 2 pipelines,  we will have to wait for both the results to complete and construct the respective summary.md file. The last step would stich the two mark down files and upload it in "Extensions" tab.

**Comment by kunalspathak** [comment]
@dotnet/jit-contrib

**Comment by JulieLeeMSFT** [comment]
Good idea. Let's target to do it in .NET 7. Assigning to @jakobbotsch now.

**Comment by BruceForstall** [comment]
There are, of course, downsides to unifying the pipelines:
1. Then you can't run just one or the other. Does that matter? Maybe only if you are developing one of the two pipelines?
2. The work involved in doing the merging, and ensuring the results are useful (as described above); a one-time amount of work

Do we have enough experience already with these to know whether unifying is better than leaving them separate?

**Comment by kunalspathak** [comment]
> Then you can't run just one or the other. Does that matter? Maybe only if you are developing one of the two pipelines?

Yes, that shouldn't matter. If you are developing one of the two pipelines, you can just comment the invocation of other pipeline in yml file. Most of the people don't care if both tools are running. In fact, it is a good thing so we can uncover bugs sooner through both the tools.

> The work involved in doing the merging, and ensuring the results are useful (as described above); a one-time amount of work

I don't think it is that big. We just need to add a leg that will wait for both Fuzzlyn and Antigen to complete, read the respective summarized md file, stitch it together and set the `uploadFile` extension variable.

**Comment by jakobbotsch** [comment]
Will push this to 8.0.
I think one thing that stops me from wanting to unify them is that Antigen does not surface reduced examples -- you have to go digging for them. If Antigen surfaced these in a nicer way I would be much more inclined to run it.

**Comment by kunalspathak** [comment]
> Antigen does not surface reduced examples 

Agree. I need to find some time to do much needed perf improvements to some of the aspects.

**Comment by jakobbotsch** [comment]
I won't have time to do this in 8.0.
